packages/access-client/src/agent-data.js

```diff
@@ -189,3 +195,39 @@ export function getSessionProofs(data) {
   }
   return proofs
 }
+
+/**
+ * Given a UCAN, return whether it is a session proof matching some options
+ *
+ * @param {Ucanto.Delegation} ucan
+ * @param {object} options
+ * @param {Ucanto.DID} [options.issuer] - issuer of session proof
+ * @param {import('@ucanto/interface').UCANLink} [options.attestedProof] - CID of proof that the session proof attests to
+ * @param {boolean} [options.allowExpired]
+ * @returns {boolean} whether the ucan matches the options
+ */
+export function matchSessionProof(ucan, options) {
```

Comment:
Where is this used?

Reply:
Originally I used this in the first thing I tried to get things to work, which was adding some session proof awareness to the agent's `.#delegations()` method (called by `.proofs()` method). Then I realized that was going to be a bit redundant because the `.proofs()` method impl calls `getSessionProofs`, so I refactored a bit to add the issuer-checking logic to `getSesssionProofs`.

This feels like a function we'll eventually want around for some advanced proof-finding scenarios, but if if I don't need it to get the tests passing, I can remove it and keep it in my notes

Reply:
I removed it